src/Benchmarks/Program.cs

```diff
@@ -133,6 +140,21 @@ private static int GetThreadCount(IConfigurationRoot config)
             var threadCountValue = config["threadCount"];
             return threadCountValue == null ? -1 : int.Parse(threadCountValue);
         }
+
+        private static IConnectionFilter GetConnectionFilter(IConfigurationRoot config, IConnectionFilter prevFilter)
```

Comment:
I don't understand what you think is unintended.  If the `--connectionFilter` command-line arg is set, then we create an instance of the type, passing either the previous connection filter or the special `NoOpConnectionFilter`.  If the `--connectionFilter` command-line arg is not set, then `options.ConnectionFilter` is not changed at all.